frontend/src/features/mapSideBar/containers/MapSideBarContainer.tsx

```diff
@@ -139,44 +139,62 @@ const MapSideBarContainer: React.FunctionComponent<IMapSideBarContainerProps> =
 
   const parcelLayerService = useLayerQuery(PARCELS_LAYER_URL);
 
-  const withNameSpace: Function = (fieldName: string, nameSpace?: string) => {
-    return nameSpace ? `${nameSpace}.${fieldName}` : fieldName;
+  /**
+   * Populate the formik form using the passed parcel.
+   * @param nameSpace the formik namespace that should be used to write any retrieved data.
+   * @param matchingParcel the parcel to use to populate the formik form.
+   */
+  const formikParcelDataPopulateCallback = (
+    matchingParcel: IParcel & ISearchFields,
+    nameSpace?: string,
+  ) => {
+    if (!formikRef.current) return;
+    if (matchingParcel.propertyTypeId !== PropertyTypes.PARCEL) {
+      toast.error('That address/pid is already in use within a subdivision.');
+      return;
+    }
+    const { setValues, values } = formikRef.current;
+    matchingParcel.propertyTypeId = getIn(values, withNameSpace(nameSpace, 'propertyTypeId'));
+    matchingParcel.parcels = getIn(values, withNameSpace(nameSpace, 'parcels'));
+    matchingParcel.searchPid = getIn(values, withNameSpace(nameSpace, 'searchPid'));
+    matchingParcel.searchPin = getIn(values, withNameSpace(nameSpace, 'searchPin'));
+    matchingParcel.searchAddress = getIn(values, withNameSpace(nameSpace, 'seachAddress'));
+    matchingParcel.evaluations = getMergedFinancials(
+      matchingParcel.evaluations,
+      Object.values(EvaluationKeys),
+    );
+    matchingParcel.fiscals = getMergedFinancials(matchingParcel.fiscals, Object.values(FiscalKeys));
+    setValues(setIn(values, nameSpace ?? '', matchingParcel));
+    toast.dark('Found matching parcel within PIMS. Form data will be pre-populated.', {
+      autoClose: 7000,
+    });
   };
 
   /**
    * Attempt to fetch the parcel within PIMS matching the passed pid or pin value. If that request fails, make another request to the parcel layer with the same data.
-   * @param pidOrPin
-   * @param parcelLayerSearchCallback
-   * @param nameSpace
+   * @param pidOrPin an object containing the pid and/or the pin
+   * @param parcelLayerSearchCallback a callback that will be executed if there is not match within PIMS for the pidOrPin
+   * @param nameSpace the formik namespace that should be used to write any retrieved data.
    */
   const fetchPimsOrLayerParcel = (
     pidOrPin: any,
     parcelLayerSearchCallback: () => void,
     nameSpace?: string,
+    formikDataPopulateCallback: (
```

Comment:
by default fetchParcelsDetail will try and populate the form pointed to by formikRef, but allow this to be overriden if necessary.